Onboarding codes added to factory data builds, weather station factory data sample aligned (#10750)
nrfconnect/sdk-nrf: applications/matter_weather_station/README.rst

Generating factory data

> =======================
> To build with custom certificates, you need to use the chip-cert tool. To do it follow the first step of the :ref:`ug_matter_device_configuring_cd_generating_steps` instruction and add the chip-cert tool to PATH when built.

Reviewer: ```suggestion
=======================

To build with custom certificates, you need to use the chip-cert tool.
Follow the first step of the :ref:`ug_matter_device_configuring_cd_generating_steps` instruction and add the chip-cert tool to PATH when built.
```
Just some style edits. You also need an empty line after the header, otherwise the doc build will create a monster.

Generating factory data
=======================

> ..  _instruction: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/matter/end_product/configuring_cd.html#id8
> 
> To build with custom certificates you need to have ``chip-cert`` tool. To do it follow the first step of the instruction_ and add ``chip-cert`` to path when built.

Reviewer: ```suggestion
To build with custom certificates, you need to use the chip-cert tool. To do it follow the first step of the :ref:`ug_matter_device_configuring_cd_generating_steps` instruction and add the chip-cert tool to PATH when built.
```

.. parsed-literal::
   :class: highlight

   west build -b thingy53_nrf5340_cpuapp -- -DOVERLAY_CONFIG=overlay-factory_data.conf

This command builds the application with default certificates.
After building the target, the generated :file:`factory_data.hex` file will be merged with the application target HEX file, so you can use the regular command to flash it to the device:

.. parsed-literal::
   :class: highlight

   west flash --erase

> If you want to use Vendor ID, Product ID or other data that is not reserved for tests, you need custom certificates.

Reviewer 2: I will change name from `custom certificates` to `custom test certificates` or `not final custom certificates`
Basically the device maker should use their of external PKI infrastructure to generate final PKI certificates for their production

   west flash --erase

> If you want to change something in your sample i. e. vendor id, product id that are not reserved for tests, you need custom certificates.
> To build with custom certificates, you need to use the chip-cert tool.
> Follow the first step of the :ref:`ug_matter_device_configuring_cd_generating_steps` instruction and add the chip-cert tool to PATH when built.

Reviewer: ```suggestion
If you want to use Vendor ID, Product ID or other data that is not reserved for tests, you need custom certificates.
To build with custom certificates, you need to use the chip-cert tool.
Follow the first step of the :ref:`ug_matter_device_configuring_cd_generating_steps` instruction and add the chip-cert tool to PATH when built.
```
